### DISCUSSION TITLE
Cannot remove clipping mask

### DISCUSSION THREAD

**Discussion by hn24** [dlemstra/Magick.NET#789] [Help] [answered]
Hi, not sure it's a bug, but im going to remove my background from my tif file using clipping path.

I found the Windows command from the [here ](https://www.imagemagick.org/discourse-server/viewtopic.php?f=27&t=34174)and it worked fine:
`magick identify -quiet -format "%[8BIM:1999,2998:#1]" image.jpg |^
magick -quiet image.jpg ( - -alpha off -negate -morphology erode diamond:1 -blur 0x1 -level 50x100% ) ^
-alpha off -compose copy_opacity -composite ^
result.png`

Also, someone has the[ same question](https://www.imagemagick.org/discourse-server/viewtopic.php?f=27&t=34174) with me, the code look like does the extract thing that the command do, but it's not work
```
using (MagickImage image = new MagickImage(bytes))
            {
                using (MagickImage image2 = new MagickImage(image.Clone()))
                {
                    image2.Alpha(AlphaOption.Off);
                    image2.Negate();
                    image2.Morphology(MorphologyMethod.Erode, Kernel.Diamond, 1);
                    image2.Blur();
                    image2.Level(new Percentage(50), new Percentage(100));

                    image.Alpha(AlphaOption.Off);
                    image.Composite(image2, CompositeOperator.CopyAlpha);

                    image.Write(AppDomain.CurrentDomain.BaseDirectory + "\\Image.png");
                }
            }
```

Just want to convert it to C#, can you help me?
My image: [my-image.zip](https://github.com/dlemstra/Magick.NET/files/5574653/my-image.zip)
My expected image:
![result](https://user-images.githubusercontent.com/53258318/99822128-69c38c80-2b85-11eb-9572-5af54c23dbee.png)

**Comment by dlemstra** [comment, marked as answer]
You are very close but the difference here is that you are not reading the clipping path when you clone it. Your command would translate to this:

```C#
using (var image = new MagickImage("image.jpg"))
{
    // magick identify -quiet -format "%[8BIM:1999,2998:#1]" \
    var clippingPathData = Encoding.ASCII.GetBytes(image.GetClippingPath());

    // ( - <= The '-' means the image is read from stdin
    using (var clippingPath = new MagickImage(clippingPathData))
    {
        // -alpha off
        clippingPath.Alpha(AlphaOption.Off);

        // -negate
        clippingPath.Negate();

        // -morphology erode diamond:1
        clippingPath.Morphology(MorphologyMethod.Erode, Kernel.Diamond, 1);

        // -blur 0x1
        clippingPath.Blur(Channels.All);

        // -level 50x100% )
        clippingPath.Level(new Percentage(50), new Percentage(100));

        // -alpha off
        image.Alpha(AlphaOption.Off);

        // -compose copy_opacity -composite
        image.Composite(clippingPath, CompositeOperator.CopyAlpha);

        image.Write("result.png");
    }
}
```

**Reply by hn24** [reply]
thanks you, it works exactly what i want.
